### PR TITLE
fzf: add compatibility with fzf≥0.48.0

### PR DESCRIPTION
**Pull request by marsam**
### Description
[fzf 0.48.0](https://github.com/junegunn/fzf/releases/tag/0.48.0) changed the way it integrates with shells.

Fixes https://github.com/nix-community/home-manager/issues/5242

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/programs/fzf.nix [hunk 1]
```diff
@@ -10,6 +10,36 @@ let
     concatStringsSep ","
     (mapAttrsToList (name: value: "${name}:${value}") colors);
 
+  hashShellIntegrationEmbedded =
```
Comment (andresilva) [inline modules/programs/fzf.nix:13]:
```suggestion
  hasShellIntegrationEmbedded =
```

File: modules/programs/fzf.nix [hunk 2]
```diff
@@ -10,6 +10,36 @@ let
...
+    lib.versionAtLeast cfg.package.version "0.48.0";
+
+  bashIntegration = if hashShellIntegrationEmbedded then ''
```
Comment (andresilva) [inline modules/programs/fzf.nix:16]:
```suggestion
  bashIntegration = if hasShellIntegrationEmbedded then ''
```

File: modules/programs/fzf.nix [hunk 3]
```diff
@@ -10,6 +10,36 @@ let
...
+      eval "$(${getExe cfg.package} --bash)"
+    fi
+  '' else ''
+    if [[ :$SHELLOPTS: =~ :(vi|emacs): ]]; then
+      . ${cfg.package}/share/fzf/completion.bash
+      . ${cfg.package}/share/fzf/key-bindings.bash
+    fi
+  '';
+
+  zshIntegration = if hashShellIntegrationEmbedded then ''
```
Comment (andresilva) [inline modules/programs/fzf.nix:27]:
```suggestion
  zshIntegration = if hasShellIntegrationEmbedded then ''
```

File: modules/programs/fzf.nix [hunk 4]
```diff
@@ -10,6 +10,36 @@ let
...
+      eval "$(${getExe cfg.package} --zsh)"
+    fi
+  '' else ''
+    if [[ $options[zle] = on ]]; then
+      . ${cfg.package}/share/fzf/completion.zsh
+      . ${cfg.package}/share/fzf/key-bindings.zsh
+    fi
+  '';
+
+  fishIntegration = if hashShellIntegrationEmbedded then ''
```
Comment (andresilva) [inline modules/programs/fzf.nix:38]:
```suggestion
  fishIntegration = if hasShellIntegrationEmbedded then ''
```

**Comment by rycee** [comment]
Thanks! Looks good to me and should be mergeable once the typo pointed out by @andresilva is resolved 🙂

**Review by andresilva — APPROVED** [review]

**Comment by rycee** [comment]
Merged now 🙂

**Pull request merged**